File: samples/vulnerable-app/js_app/crypto_demo.js
// Intentionally uses classical crypto for scanner demo (not production).
// pqc-inventory-file: owner=sample-js-team
const crypto = require('crypto');
const forge = require('node-forge');
const { SignJWT } = require('jose');

// Node RSA — high risk
// pqc-inventory: lifetime=12 exposure=public_key owner=key-mgmt
crypto.generateKeyPairSync('rsa', { modulusLength: 2048 });

// Node EC / ECDH — high
crypto.generateKeyPairSync('ec', { namedCurve: 'P-256' });
const ecdh = crypto.createECDH('prime256v1');

// Ed25519 — high (classical EdDSA)
crypto.generateKeyPairSync('ed25519');

// node-forge RSA — high
const keys = forge.pki.rsa.generateKeyPair({ bits: 2048 });

// jose classical JWT alg — high (trust-boundary exposure override)
// pqc-inventory: lifetime=10 exposure=trust_boundary owner=auth
const alg = 'RS256';

// WebCrypto-style algorithm names
const rsaAlg = 'RSA-OAEP';
const ecAlg = { name: 'ECDSA', namedCurve: 'P-256' };

// AES — low
const cipher = crypto.createCipheriv('aes-256-gcm', Buffer.alloc(32), Buffer.alloc(12));

// Hash — info (local checksum noise; suppressed by default hash-policy)
const hash = crypto.createHash('sha256').update('demo').digest('hex');

File: samples/vulnerable-app/python_app/crypto_demo.py
"""Intentionally uses classical crypto APIs for scanner demo (not production)."""

# pqc-inventory-file: owner=sample-python-team

import hashlib
import ssl
from cryptography.hazmat.primitives.asymmetric import rsa, ec, ed25519
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
import ecdsa

# RSA key generation — quantum high risk (Shor)
# Long-lived payment material: explicit lifetime + exposure override (demo).
# pqc-inventory: lifetime=15 exposure=public_key owner=payments
private_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)

# Classical ECDSA / ECDH
# pqc-inventory: lifetime=8 exposure=public_key
ec_key = ec.generate_private_key(ec.SECP256R1())

# Ed25519 classical signature
ed_key = ed25519.Ed25519PrivateKey.generate()

# python-ecdsa
sk = ecdsa.SigningKey.generate(curve=ecdsa.NIST256p)

# TLS wrapper — medium (check suites / PQC readiness)
# pqc-inventory: lifetime=5 exposure=trust_boundary owner=edge-tls
ctx = ssl.create_default_context()

# AES — low (prefer 256-bit under Grover)
aes_cipher = Cipher(algorithms.AES(b"0" * 32), modes.ECB())

# Hash integrity — info (no lifetime override; remains unknown)
digest = hashlib.sha256(b"demo").hexdigest()
